src/praisonai-bot/praisonai_bot/gateway/supervisor.py

```
@@ -232,13 +232,24 @@ async def run(self, name: str, bot: Any, start_fn: Callable[[str, Any], Any]) ->
                 bot_task = asyncio.create_task(start_fn(name, bot))
                 abort_task = asyncio.create_task(abort_signal.wait())
                 
+                # Issue #4043: the channel is coming back after a transient
+                # outage. Re-attempt anything the durable outbox held through the
+                # outage now, instead of leaving it undelivered until the next
+                # inbound chat() turn or a process restart. Scheduled as a
+                # background task (not awaited here) so it runs once the adapter
+                # has reconnected without blocking the supervision wait — bounded
+                # and best-effort, see _on_channel_recovered.
+                if recovering:
+                    asyncio.create_task(self._on_channel_recovered(name, bot))
```

Ruff (0.16.1)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)